### DISCUSSION TITLE
Bad Wait for initial Data OPC UA Python

### DISCUSSION THREAD

**Discussion by AdhamBayoumi** [community/community#31880] [Other Feature Feedback, Questions, & Ideas]
I've a problem with a 3rd party Server that I can't get the value of the node cause error of Bad wait for initial data ,I've searched and found that I should use the Subscription I did but the problem is still founded , FYI when i use UAExpert Every things goes fine i could read the value 
here is my code for help Thanks,  
 
import sys
import time

from opcua import Client, Subscription
from opcua import ua

class SubHandler(object):
      def data_change(self, handle, node, val, attr):
        pass
    def datachange_notification(self, node, val, data):
        pass
    def event_notification(self, event):
        pass
    def status_change_notification(self, status):
        pass

if __name__ == "__main__":
    client = Client("opc.tcp://10.3.238.234:55000")
    params = ua.CreateSubscriptionParameters()
    params.RequestedPublishingInterval = 1000
    params.RequestedLifetimeCount = 3000
    params.RequestedMaxKeepAliveCount = 10000
    params.MaxNotificationsPerPublish = 10000
    params.PublishingEnabled = True
    params.Priority = 0

    client.session_timeout = 120000
    client.secure_channel_timeout = 3600000
    client.connect()
    print("OPCUA is connected to server")
       
while True:
    try:

        client.load_type_definitions()
        root = client.get_root_node()
        print("Root node is: ", root)
        objects = client.get_objects_node()
        print("Objects node is: ", objects)
        print("Children of root are: ", root.get_children())
        var = client.get_node("ns=1;s=OCTAGON.Satellites.S5-03.DataPacks.Irrigation_Queue.StationStates")
        print(var)
        handler =SubHandler()
        sub = client.create_subscription(5000, handler)
        print("OPCUA Create Subscription")
        handle = sub.subscribe_data_change(var)
        hand=sub.create_monitored_items(var)
        tempT1 = root.get_child(["0:Objects", "1:OCTAGON.Satellites.S5-03.DataPacks.Irrigation_Queue.StationStates"])
        print (tempT1)
        print(var)
        b = var.get_data_value()  # get value of node as a DataValue object
        c = var.get_value()
        print("my-var is: ", b, c)
       
        
        print(handle)

          # get value of node as a python builtin
        time.sleep(10)

    except  :
        sub.unsubscribe()
        sub = client.create_subscription(5000, handler)
        print (sub)
        time.sleep(20)

**Comment by AdhamBayoumi** [comment]
@zackfern   @davidstaheli   @courtneycl   @jeffbeehler   
Anyone Could help Please

**Comment by queenofcorgis** [comment]
Hi @AdhamBayoumi,

Thanks for being a part of the GitHub Community! We're glad you're here.

I wanted to jump in and mention that if you're looking for help for this specific topic, you might want to try asking for help somewhere that focuses on that server. It's definitely possible that another GitHub user might have run into this same issue and can help, but the GitHub Community on Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck!

**Comment by AdhamBayoumi** [comment]
Thanks for your consideration
I really do appreciate it but I cannot figure Where Should I request my
issue
Could you help ?
Thanks
